### PR TITLE
Add haproxy role

### PR DESCRIPTION
**Pull request by pmaslana**

### CODE REVIEW

File: haproxy/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+haproxy-host-name-1: ""
+haproxy-host-name-2: ""
+haproxy-host-name-3: ""
+
+haproxy-host-ip-1: ""
+haproxy-host-ip-2: ""
+haproxy-host-ip-3: ""
```
Comment (cmmarslender) [inline haproxy/defaults/main.yml:7]: These should be lists and then loop in the template so its the most flexible

File: haproxy/templates/haproxy.cfg.j2 [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    mode tcp
+    default_backend kubeAPI_backend
+frontend konnectivity
+    bind :8132
+    mode tcp
+    default_backend konnectivity_backend
+frontend controllerJoinAPI
+    bind :9443
+    mode tcp
+    default_backend controllerJoinAPI_backend
```
Comment (cmmarslender) [inline haproxy/templates/haproxy.cfg.j2:47]: Since this is so specific to k0s, might be worth changing the role name to haproxy-k0s or something

**Review by cmmarslender — CHANGES_REQUESTED** [review]

File: haproxy/defaults/main.yml (hunk 1 above)
Comment (pmaslana) [reply haproxy/defaults/main.yml]: Created list and loop

File: haproxy/templates/haproxy.cfg.j2 (hunk 2 above)
Comment (pmaslana) [reply haproxy/templates/haproxy.cfg.j2]: renamed the file

**Review by cmmarslender — APPROVED** [review]

**Pull request merged**